scrapy/utils/spider.py

```diff
@@ -18,7 +18,11 @@ def iterate_spider_output(result):
         d = deferred_from_coro(collect_asyncgen(result))
         d.addCallback(iterate_spider_output)
         return d
-    return arg_to_iter(deferred_from_coro(result))
+    elif inspect.iscoroutine(result):
+        d = deferred_from_coro(result)
+        d.addCallback(iterate_spider_output)
+        return d
+    return arg_to_iter(result)
```

Comment:
It's safe to remove the `deferred_from_coro` call here, because at this point [this condition](https://github.com/scrapy/scrapy/blob/2.1.0/scrapy/utils/defer.py#L135) is `False` so the passed object is returned unmodified.